src/App.tsx

```diff
@@ -1,24 +1,244 @@
+
+      setTodos(response);
+    } catch (error) {
+      setErrorType(ErrorTypes.UPLOAD);
+    }
+  };
+
+  const deleteTodoFromServer = useCallback(async (todoId: number) => {
+    setRemovingTodoIds(prevTodoIds => [...prevTodoIds, todoId]);
+
+    try {
+      await deleteTodo(todoId);
+
+      const updatedTodos = todos.filter(todo => todo.id !== todoId);
+
+      setTodos(updatedTodos);
+    } catch (error) {
+      setErrorType(ErrorTypes.DELETE);
+    } finally {
+      setRemovingTodoIds(prevTodoIds => prevTodoIds
+        .filter((id) => id === todoId));
+    }
+  }, [todos]);
+
+  const clearCompletedTodos = useCallback(async () => {
+    setRemovingTodoIds(completedTodosIds);
+
+    Promise.all(completedTodosIds.map((id) => deleteTodo(id)))
+      .then(() => {
+        setTodos(activeTodos);
+      })
+      .catch(() => {
+        setErrorType(ErrorTypes.DELETE);
+      })
+      .finally(() => setRemovingTodoIds(prevTodoIds => prevTodoIds
+        .filter((id) => !completedTodosIds.includes(id))));
+  }, [hasCompletedTodo, activeTodos]);
+
+  const changeTodoStatus = useCallback(async (todo: Todo) => {
+    setUpdatingTodoIds(prevTodoIds => [...prevTodoIds, todo.id]);
+
+    try {
+      const updatedTodo = await updateTodo(todo.id, {
+        ...todo,
+        completed: !todo.completed,
+      });
+
+      const updatedTodos = todos.map(todoItem => {
+        if (todoItem.id === updatedTodo.id) {
+          return updatedTodo;
+        }
+
+        return todoItem;
+      });
+
+      setTodos(updatedTodos);
+    } catch (error) {
+      setErrorType(ErrorTypes.UPDATE);
+    } finally {
+      setUpdatingTodoIds(prevTodoIds => prevTodoIds
+        .filter((id) => id !== todo.id));
+    }
+  }, [activeTodosIds, completedTodosIds]);
+
+  const changeTodoTitle = useCallback(async (
+    todo: Todo,
+    newTitle: string,
+  ) => {
+    setUpdatingTodoIds(prevTodoIds => [...prevTodoIds, todo.id]);
+
+    try {
+      const updatedTodo = await updateTodo(todo.id, {
+        ...todo,
+        title: newTitle,
+      });
+
+      const updatedTodos = todos.map(todoItem => {
+        if (todoItem.id === updatedTodo.id) {
+          return updatedTodo;
+        }
+
+        return todoItem;
+      });
+
+      setTodos(updatedTodos);
+    } catch (error) {
+      setErrorType(ErrorTypes.UPDATE);
+    } finally {
+      setUpdatingTodoIds(prevTodoIds => prevTodoIds
+        .filter((id) => id !== todo.id));
+    }
+  }, [title, updatingTodoIds, todos]);
+
+  useEffect(() => {
+    getTodosFromServer();
+  }, []);
+
+  const toggleAll = useCallback(async () => {
+    const updateActive = () => {
+      return todos.map(todo => {
+        setUpdatingTodoIds(activeTodosIds);
+
+        if (!todo.completed) {
+          return updateTodo(todo.id, {
+            ...todo,
+            completed: !todo.completed,
+          });
+        }
+
+        return todo;
+      });
+    };
+
+    const updateCompleted = () => {
```

Comment:
```suggestion
    const handleUpdateCompleted = () => {
```

Reply:
Thank you  for improves, changed naming convention in this case. In addition to this i fix problem with todo status change, function filtered todos was with error(in case all - showed not completed todos) Finally, problem in changing title to '' - is not a problem, cause the task said to do this in this way, even standart placeholder say - 'Empty todo will be deleted'.